### PR TITLE
feat: Show prompt to login when follow group clicked on promoted group

### LINKED ISSUE #8070: Groups: Add follow button for not logged in users and option to log in
Users who are not logged in should see a kind of "inactive" follow button on groups pages similar to the front page "whitened" follow button.

When the user tries to click on the button they should receive a pop up with the following text:
"You need to be logged in to follow this group. Login / Cancel"

Compare https://eventyay.com/g/40

Group page with right-side area missing "Following" button.

Screenshot from 2021-11-11 19-50-27

Button is visible when logged in.

Screenshot from 2021-11-11 19-50-47

"Whitened" button on frontpage for not logged in users

Screenshot from 2021-11-11 19-55-14

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #8070 

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/templates/components/promoted-group-card.hbs [hunk 1]
```diff
@@ -13,7 +13,7 @@
         </span>
       </div>
     </span>
-    <button class="blue {{if (not this.authManager.currentUser.email) 'disabled'}} {{if this.device.isMobile 'mt-2' 'mt-4'}} ui fluid button" {{action 'follow'}}>
```
Comment (pc-beast) [inline app/templates/components/promoted-group-card.hbs:16]: Why remove this. This will remove the inactive color from the button while not logged in.

Comment (codedsun) [reply app/templates/components/promoted-group-card.hbs:16]: We have to add alpha to the color of the button as discussed in meeting @pc-beast .

**Comment by codedsun** [comment]
image

File: app/templates/components/forms/group/group-view.hbs [hunk 2]
```diff
@@ -69,13 +69,12 @@
       {{/if}}
     </div>
     <div class="three wide column">
-      {{#if this.session.isAuthenticated}}
-        <button class="blue {{if this.device.isMobile 'mt-2' 'mt-4'}} ui fluid button" {{action 'toggleFollow'}}>
+        <button class="blue {{if this.device.isMobile 'mt-2' 'mt-4'}} ui fluid button" style="background-color:rgba(65, 150, 214, 0.40)" {{action 'toggleFollow'}}>
```
Comment (progmatic-99) [inline app/templates/components/forms/group/group-view.hbs:72]: We only want the alpha when the user is **logged out**. Also, the style is repeating and will be used at more places regarding groups, so we can create a class for this.

File: app/templates/components/promoted-group-card.hbs [hunk 3]
```diff
@@ -13,7 +13,7 @@
...
+    <button class="blue {{if this.device.isMobile 'mt-2' 'mt-4'}} ui fluid button" style="background-color:rgba(65, 150, 214, 0.40)" {{action 'follow'}}>
```
Comment (progmatic-99) [inline app/templates/components/promoted-group-card.hbs:16]: Same as above.

**Review by progmatic-99 — CHANGES_REQUESTED** [review]

File: app/templates/components/group-card.hbs [hunk 4]
```diff
@@ -60,7 +60,7 @@
             {{#if this.isFollowed}}
                 {{t 'Following'}}
             {{else}}
-                {{t 'Follow Group'}}         
+                {{t 'Follow group Group'}}
```
[the remark this answers is not in the document]
Comment (codedsun) [reply app/templates/components/group-card.hbs]: this has been outdated! @progmatic-99

File: app/templates/components/forms/group/group-view.hbs (hunk 2 above)
Comment (codedsun) [reply app/templates/components/forms/group/group-view.hbs:72]: So should I create two buttons?

Comment (progmatic-99) [reply app/templates/components/forms/group/group-view.hbs:72]: No

**Review by pc-beast — CHANGES_REQUESTED** [review]
Changes remaining on explore groups page and event page.

**Review by pc-beast — APPROVED** [review]

**Review by progmatic-99 — APPROVED** [review]

**Pull request merged**